docs/Integrating Your HMI/Developing the HMI UI/index.md

```diff
@@ -0,0 +1,445 @@
+
+For each app listed in the `UpdateAppList` request, the HMI's App List should show a button that includes the app's name and icon.
+
+![App List](./assets/AppList1.png)
+
+If an app is disconnected or unregistered, Core will send an `UpdateAppList` request to the HMI with that application omitted from the app list array.
+
+The HMI should make sure its app list is always up to date, and only show applications that were included in the most recent `UpdateAppList` request.
+
+## Activating an Application
+
+### User Selection
+
+When the user selects an application from the app list, a request should be made to bring this app to the foreground (this is called "activating" the application). The first step required by the HMI when an application is selected is to send a `SDL.ActivateApp` request to Core.
+
+When Core responds with a successful `SDL.ActivateApp` response, the HMI can switch views from the app list to the app's default template.
+
+!!! NOTE
+The default template for an app should be used if the app has not requested to use a specific template via the `UI.Show.templateConfiguration` parameter or `UI.SetDisplayLayout` (deprecated).
+
+The default template for media apps is `MEDIA`, and the default template for all other apps should be `NON-MEDIA`.
+
+You can check a given app is a media application using that app's `isMediaApplication` parameter, sent in the `BasicCommunication.UpdateAppList` request.
+!!!
+
+### User Consent
+
+If the `SDL.ActivateApp` response returns with the parameter `isPermissionsConsentNeeded = true`, the HMI should send a `SDL.GetListOfPermissions` request. This happens when the activating app requires permissions that the user must provide consent for. For example, if an app wants to access vehicle data, an SDL policy configuration might require that the user consent to allow the app to collect this information.
+
+After receiving the list of permissions for the app, the HMI should show the user the `PermissionItem` name and status for each requested permission. The user should have the ability to enable or disable each permission item. If any permission changes are made by the user, these updates should be communicated to core via `SDL.OnAppPermissionConsent` notification.
+
+!!! NOTE
+Permissions are managed by SDL Core's policy table. Refer to the [SDL Overview Policy Guide](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/overview/).
+!!!
+
+### Resumption
+
+If an app is disconnected from SDL Core and reconnects within a specified time limit, Core will try to resume the app into the same HMI state the app was in before resumption. The HMI should be prepared to handle a `BasicCommunication.ActivateApp` request from SDL Core, in which case the HMI should return the app into the requested state (or respond with an error if unable to). For example, if the requested HMI level is `FULL`, the HMI should activate the app and put that app's template into view.
```

Comment:
>If an app is disconnected from SDL Core and reconnects within a specified time limit, Core will try to resume the app into the same HMI state the app was in before resumption. 

Should this instead end "...into the same HMI state the app was in before it was disconnected."?